src/convert.ts

```diff
@@ -1,64 +1,19 @@
-import { FileCollection, groupFiles } from 'filelist-utils';
+import type { BinaryData } from 'cheminfo-types';
 
-import { ComplexObject } from './Types';
-import { MPR, parseMPR } from './mpr/parseMPR';
+import { parseMPR } from './mpr/parseMPR';
 import { parseMPS } from './mps/parseMPS';
-import { MPT, parseMPT } from './mpt/parseMPT';
+import { parseMPT } from './mpt/parseMPT';
 
-/**
- * Text files have no type at the moment, but they
- * follow MPR as much as possible
- */
-export interface Biologic {
-  dir?: string;
-  mpr?: MPR;
-  mps?: ComplexObject;
-  mpt?: MPT;
-}
-
-/**
- *  Parses Biologic mpr, mpt, mps formats from multiple or single directories.
- *  The result contains an array of [[`Biologic`]] `[{dir:'1', mps, mpt}, {dir:'2', mps, mpt},..]`
- *
- *  Project structure example:
- *
- *  ```text
- *├── parent
- *│  ├── child1
- *│  │  ├── jdb11-1.mpr
- *│  │  └── jdb11-1.mps
- *   ...
- *│  └── childN
- *│      ├── test.mpr
- *│      ├── test.mps
- *│      └── test.mpt
- *  ```
- *
- * @param fileCol - `path/to/parent` or `path/to/any/child`. (See tree above.)
- * @returns JSON object passing **child** directory; array of children if you pass a **parent**.
- */
-
-export async function convert(fileCol: FileCollection): Promise<Biologic[]> {
-  const dirs = groupFiles(fileCol);
-  let results: Biologic[] = [];
-
-  /* can not use `forEach` and pass `async` functions */
-  for (const dir of dirs) {
-    let result: Biologic = {};
-    result.dir = dir.key;
-    for (const dataFile of dir.fileCollection) {
-      const fName = dataFile.name;
-      if (fName.endsWith('.mps')) {
-        result.mps = parseMPS(await dataFile.arrayBuffer());
-      } else if (fName.endsWith('.mpt')) {
-        result.mpt = parseMPT(await dataFile.arrayBuffer());
-      } else if (fName.endsWith('.mpr')) {
-        result.mpr = parseMPR(await dataFile.arrayBuffer());
-      }
-    }
-    if (Object.keys(result).length > 1) {
-      results.push(result);
-    }
+export type FileType = 'mpr' | 'mpt' | 'mps';
```

Comment:
In the current implementation there is no advantage to have the method 'convert' so I think we can simply remove it as well and keep only parseMPS, parseMPT and parseMPR

The alternative is to be able to convert the 3 formats at once and have something like

`convert({mpr?, mps?, mpt?}) that returns an object {mpr, mps, mpt}`

Reply:
I agree the `convert` function has an api which makes it more difficult to use than using the individual parsers directly. It can be removed.

Reply:
Will update it, Thanks!